### PR TITLE
Added code to merge two trees

### PR DESCRIPTION
**Pull request by jindal2309**
### **Describe your change:** 

**Merge two binary trees**
*Problem Description:* Given two binary tree, return the merged tree.
The rule for merging is that if two nodes overlap, then put the value sum of
both nodes to the new value of the merged node. Otherwise, the NOT null node
will be used as the node of the new tree.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by ruppysuppy — CHANGES_REQUESTED** [review]
Doctest for `main` is irrelevant, please implement all other suggestions made by `algorithms-keeper`

File: data_structures/binary_tree/merge_two_binary_trees.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    >>> print_preorder(root.right)
+    3
+    """
+    if root:
+        print(root.value)
+        print_preorder(root.left)
+        print_preorder(root.right)
+
+
+def main() -> None:
```
[the remark this answers is not in the document]
Comment (jindal2309) [reply data_structures/binary_tree/merge_two_binary_trees.py]: Done.

File: data_structures/binary_tree/merge_two_binary_trees.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
```
[the remark this answers is not in the document]
Comment (jindal2309) [reply data_structures/binary_tree/merge_two_binary_trees.py]: Done

File: data_structures/binary_tree/merge_two_binary_trees.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    A binary node has value variable and pointers to its left and right node.
+    """
+
+    def __init__(self, value=0, left=None, right=None):
+        self.value = value
+        self.left = left
+        self.right = right
+
+
+def merge_two_binary_trees(tree1: Node, tree2: Node) -> Node:
```
[the remark this answers is not in the document]
Comment (jindal2309) [reply data_structures/binary_tree/merge_two_binary_trees.py]: Done

**Review by ruppysuppy — APPROVED** [review]

**Pull request merged**